video/out/vulkan/context_display.c

```diff
@@ -0,0 +1,461 @@
+                    }
+                }
+                displays = NULL;
+            }
+        }
+
+        talloc_free(props);
+        props = NULL;
+    }
+
+done:
+    talloc_free(devices);
+    talloc_free(props);
+    talloc_free(modes);
+    talloc_free(planes);
+    talloc_free(planes_to_displays);
+    return ret;
+}
+
+static int display_validate_spec(struct mp_log *log, const struct m_option *opt,
+                                struct bstr name, struct bstr param)
+{
+    if (bstr_equals0(param, "help")) {
+        print_display_info(log);
+        return M_OPT_EXIT;
+    }
+
+    char *spec = bstrto0(NULL, param);
+    if (!parse_display_spec(log, spec, NULL, NULL, NULL)) {
+        mp_fatal(log, "Invalid value for option vulkan-display-spec. "
+                 "Must be a string of the format 'D:M:P' where each of D, M, P "
+                 "is the index of a Display, Mode, and Plane, or 'help'\n");
+        talloc_free(spec);
+        return M_OPT_INVALID;
+    }
+    talloc_free(spec);
+
+    return 1;
+}
+
+#define OPT_BASE_STRUCT struct vulkan_display_opts
+const struct m_sub_options vulkan_display_conf = {
+    .opts = (const struct m_option[]) {
+        OPT_STRING_VALIDATE("vulkan-display-spec", display_spec, 0, display_validate_spec),
+        {0}
+    },
+    .size = sizeof(struct vulkan_display_opts),
+    .defaults = &(struct vulkan_display_opts) {
+        .display_spec = "0:0:0",
+    },
+};
+
+struct priv {
+    struct mpvk_ctx vk;
+    struct vulkan_display_opts *opts;
+    uint32_t width;
+    uint32_t height;
+};
+
+static void display_uninit(struct ra_ctx *ctx)
+{
+    struct priv *p = ctx->priv;
+
+    ra_vk_ctx_uninit(ctx);
+    mpvk_uninit(&p->vk);
+}
+
+static bool display_init(struct ra_ctx *ctx)
+{
+    struct priv *p = ctx->priv = talloc_zero(ctx, struct priv);
+    struct mpvk_ctx *vk = &p->vk;
+    int msgl = ctx->opts.probing ? MSGL_V : MSGL_ERR;
+    VkResult res;
+    VkPhysicalDevice *devices = NULL;
+    VkDisplayPropertiesKHR *props = NULL;
+    VkDisplayModePropertiesKHR *modes = NULL;
+    uint32_t num = 0;
+
+    p->opts = mp_get_config_group(p, ctx->global, &vulkan_display_conf);
+    int display_idx, mode_idx, plane_idx;
+    parse_display_spec(ctx->log, p->opts->display_spec,
+                       &display_idx, &mode_idx, &plane_idx);
+
+    if (!mpvk_init(vk, ctx, VK_KHR_DISPLAY_EXTENSION_NAME))
+        goto error;
+
+    char *device_name = ra_vk_ctx_get_device_name(ctx);
+    struct pl_vulkan_device_params vulkan_params = {
+        .instance = vk->vkinst->instance,
+        .device_name = device_name,
+    };
+    VkPhysicalDevice device = pl_vulkan_choose_device(vk->ctx, &vulkan_params);
+    talloc_free(device_name);
+    if (!device) {
+        MP_MSG(ctx, msgl, "Failed to open physical device.\n");
+        goto error;
+    }
+
+    res = vkGetPhysicalDeviceDisplayPropertiesKHR(device, &num, NULL);
+    if (res != VK_SUCCESS) {
```

Comment:
FWIW these cases are basically impossible, might as well not even bother testing this return value. Just make sure to initialize `num` to 0 and nothing will go wrong even if it hypothetically fails somehow.